processor/internal/transformer/destination_transformer/embedded/warehouse/datatype.go

```diff
@@ -1,15 +1,10 @@
 package warehouse
 
 import (
-	"fmt"
```

Comment:
> We rely on being SingularEvent only to contain map[string]any, but some data fields are directly being added, like say for Tracking Plan, Bot Details and Geo enrichment.

Can you add more details about what does "some data fields are directly being added" mean ?